x/txfees/keeper/feedecorator.go

```diff
@@ -58,6 +62,40 @@ func (mfd MempoolFeeDecorator) AnteHandle(ctx sdk.Context, tx sdk.Tx, simulate b
 		}
 	}
 
+	msgs := tx.GetMsgs()
+	for _, msg := range msgs {
+		// If one of the msgs is an IBC Transfer msg, limit it's size due to current spam potential.
+		// 500KB for entire msg
+		// 400KB for memo
+		// 65KB for receiver
+		if transferMsg, ok := msg.(*transfertypes.MsgTransfer); ok {
+			if transferMsg.Size() > 500000 { // 500KB
+				return ctx, errorsmod.Wrap(sdkerrors.ErrInvalidRequest, "msg size is too large")
+			}
+
+			if len([]byte(transferMsg.Memo)) > 400000 { // 400KB
+				return ctx, errorsmod.Wrap(sdkerrors.ErrInvalidRequest, "memo is too large")
+			}
+
+			if len(transferMsg.Receiver) > 65000 { // 65KB
+				return ctx, errorsmod.Wrap(sdkerrors.ErrInvalidRequest, "receiver address is too large")
+			}
+		}
+
+		// If one of the msgs is from ICA, limit it's size due to current spam potential.
+		// 500KB for packet data
+		// 65KB for sender
+		if icaMsg, ok := msg.(*icacontrollertypes.MsgSendTx); ok {
+			if icaMsg.PacketData.Size() > 500000 { // 500KB
+				return ctx, errorsmod.Wrap(sdkerrors.ErrInvalidRequest, "packet data is too large")
+			}
+
+			if len([]byte(icaMsg.Owner)) > 65000 { // 65KB
+				return ctx, errorsmod.Wrap(sdkerrors.ErrInvalidRequest, "owner address is too large")
+			}
+		}
+	}
```

Comment:
Consider copying some of the violating message JSONs from mintscan and adding them as unit tests

There is an example with arbitrage filters to help with boilerplate setup for this

Reply:
Tests added here https://github.com/osmosis-labs/osmosis/pull/8409/commits/179d7e7d61aacedee001c6a6313455e14e6bfea6